core-web/libs/ui/src/lib/dot-tab-buttons/dot-tab-buttons.component.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Spectator, byTestId, createComponentFactory } from '@ngneat/spectator';
+
+import { CommonModule, NgClass, NgIf } from '@angular/common';
+
+import { SelectItem } from 'primeng/api';
+import { ButtonModule } from 'primeng/button';
+
+import { DotPageMode } from '@dotcms/dotcms-models';
+
+import { DotTabButtonsComponent } from './dot-tab-buttons.component';
+
+describe('DotTabButtonsComponent', () => {
+    let spectator: Spectator<DotTabButtonsComponent>;
+
+    const createComponent = createComponentFactory({
+        component: DotTabButtonsComponent,
+        imports: [CommonModule, ButtonModule, NgIf, NgClass]
+    });
+    const optionsMock: SelectItem[] = [
+        { label: 'Edit', value: DotPageMode.EDIT },
+        { label: 'Preview', value: DotPageMode.PREVIEW }
+    ];
+
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
+                options: optionsMock,
+                mode: DotPageMode.PREVIEW
+            }
+        });
+    });
+
+    it('should render options', () => {
+        spectator.detectChanges();
```

Comment:
When you create the component and not use:
```
spectator = createComponent({
            detectChanges: false
        });
```
No need to trigger `spectator.detectChanges();` every test.